gadopt/approximations.py

```diff
@@ -432,49 +518,288 @@ class SmallDisplacementViscoelasticApproximation:
+    to laterally varying viscosity. Geophysical journal international, 214(2), 1324-1363.
+
+    Automated forward and adjoint modelling of viscoelastic deformation of the solid
+    Earth.  Scott, W.; Hoggard, M.; Duvernay, T.; Ghelichkhan, S.; Gibson, A.;
+    Roberts, D.; Kramer, S. C.; and Davies, D. R. EGUsphere, 2025: 1–43. 2025.
+
+
+    Arguments:
+      bulk_modulus:  bulk modulus
+      density:       background density
+      shear_modulus: shear modulus
+      viscosity:     viscosity
+      bulk_shear_ratio: Ratio of bulk to shear modulus
+      compressible_buoyancy: Include compressible buoyancy effects
+      compressible_adv_hyd_pre: Include compressible hydrostatic prestress advection
+      g:             gravitational acceleration
+      B_mu:          Nondimensional number describing ratio of buoyancy to elastic
+                     shear strength used for nondimensionalisation.
+                     $ B_{\\mu} = \frac{\bar{\rho} \bar{g} L}{\bar{\\mu}}$,
+                     where $\bar{\rho}$ is a characteristic density scale (kg / m^3),
+                     $\bar{g}$ is a characteristic gravity scale (m / s^2),
+                     $L$ is a characteristic length scale, often Mantle depth (m),
+                     $\\mu$ is a characteristic shear modulus (Pa).
+
+    """
+
+    compressible = True
+
+    def __init__(
+        self,
+        bulk_modulus: Function | Number,
+        density: Function | Number,
+        shear_modulus: Function | Number | list[Function | Number],
+        viscosity: Function | Number | list[Function | Number],
+        *,
+        bulk_shear_ratio: Function | Number = 1,
+        **kwargs,
+    ):
+        super().__init__(
+            bulk_modulus,
+            density,
+            shear_modulus,
+            viscosity,
+            bulk_shear_ratio=bulk_shear_ratio,
+            **kwargs,
+        )
+
+    def buoyancy(self, displacement: Function) -> ufl.core.expr.Expr:
+        # Compressible part of buoyancy term due to the density perturbation
+        # written on rhs of equations
+        buoyancy = super().buoyancy(displacement)
+        # By default this term is included but in some cases e.g. to reproduce
+        # simplified analytical cases such as the Cathles 2024 benchmark in
+        # /tests/glacial_isostatic_adjustment/iv_ve_fs.py we need to remove
+        # this effect.
+        buoyancy += -self.B_mu * self.g * -self.density * div(displacement)
+        return buoyancy
+
+    def compressible_adv_hyd_pre(
+        self, u_r: Function | ufl.core.expr.Expr
+    ) -> ufl.core.expr.Expr:
+        # Hydrostatic prestress advection term which is applied
+        # as a `normal_stress` boundary condition when the `free_surface` tag is
+        # specified in the `bcs` dictionary of the `InternalVariableSolver`
+        # through the `set_free_surface_boundary` method.
+        return self.B_mu * self.density * self.g * u_r
```

Comment:
This method is identical to `hydrostatic_prestress_advection` in the parent class, but it has a different name. Why?

Reply:
It's the integrated by parts term which has the same form. I've changed it so it returns the hydrostatic prestress advection expression and updated the comment a little.